app/src/main/java/com/termux/api/apis/UsbAPI.java

```diff
@@ -212,17 +219,34 @@ protected int open(@NonNull UsbDevice device) {
 
 
 
-        protected UsbDevice getDevice(Intent intent, String deviceName) {
+        protected UsbDevice getDevice(Intent intent, @Nullable String deviceName, @Nullable String vendorId, @Nullable String productId) {
             UsbManager usbManager = (UsbManager) getSystemService(Context.USB_SERVICE);
 
-            HashMap<String, UsbDevice> deviceList = usbManager.getDeviceList();
-            UsbDevice device = deviceList.get(deviceName);
-            if (device == null) {
-                Logger.logVerbose(LOG_TAG, "Failed to find device \"" + deviceName + "\"");
-                ResultReturner.returnData(this, intent, out -> out.append("No such device.\n"));
+            if (deviceName != null) {
+                HashMap<String, UsbDevice> deviceList = usbManager.getDeviceList();
+                UsbDevice usbDevice = deviceList.get(deviceName);
+                if (usbDevice != null) return usbDevice;
             }
 
-            return device;
+            if (vendorId != null && productId != null){
+                // UsbDevice returns int, not hex, correct for that.
+                int iVendorId = Integer.parseInt(vendorId, 16);
+                int iProductId = Integer.parseInt(productId, 16);
+                for (UsbDevice usbDevice : usbManager.getDeviceList().values()) {
+                    if (usbDevice.getVendorId() == iVendorId && usbDevice.getProductId() == iProductId) {
+                        if (deviceName != null && !(deviceName.equals(usbDevice.getDeviceName()))){
```

Comment:
In this one as well.
```suggestion
                        if (deviceName != null && !(deviceName.equals(usbDevice.getDeviceName()))) {
```